client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-recipe.ts

```diff
@@ -38,105 +66,127 @@ const useRecipe = ( siteId = 0, patterns: Pattern[], categories: Category[] ) =>
 		enabled: !! font_variation_title,
 	} );
 
-	const { setSelectedDesign } = useDispatch( ONBOARD_STORE );
-
-	const selectedDesignRef = useRef< Design | undefined >( selectedDesign );
+	const colorVariation =
+		( colorVariations || [] ).find( ( { title } ) => title === color_variation_title ) || null;
+
+	const fontVariation =
+		( fontVariations || [] ).find( ( { title } ) => title === font_variation_title ) || null;
+
+	const setHeader = ( pattern: Pattern | null ) => {
+		setSearchParams(
+			( currentSearchParams ) => {
+				if ( pattern === null ) {
+					currentSearchParams.delete( 'header_pattern_ids' );
+				} else {
+					currentSearchParams.set( 'header_pattern_ids', '' + pattern.ID );
```

Comment:
I'm curious about these concatenations `'' + pattern.ID`. Is it a way to change the type to string?

Reply:
Yes, because `pattern.ID` is `number`, but `.set()` expects a `string` 🥲